docs/orchestration.asciidoc

```diff
@@ -0,0 +1,106 @@
+[id="{p}-orchestration"]
+=== NodeSets orchestration
+
+[id="{p}-nodesets"]
+==== NodeSets overview
+
+The Elasticsearch cluster is specified using a list of NodeSets. Each NodeSet represents a group of Elasticsearch nodes sharing the same specification (both Elasticsearch configuration and Kubernetes Pod configuration).
+
+[source,yaml,subs="attributes"]
+----
+apiVersion: elasticsearch.k8s.elastic.co/{eck_crd_version}
+kind: Elasticsearch
+metadata:
+  name: quickstart
+spec:
+  version: {version}
+  nodeSets:
+  - name: master-nodes
+    count: 3
+    config:
+      master: true
+      data: false
+    volumeClaimTemplates:
+    - metadata:
+        name: elasticsearch-data
+      spec:
+        accessModes:
+        - ReadWriteOnce
+        resources:
+          requests:
+            storage: 10Gi
+        storageClassName: standard
+  - name: data-nodes
+    count: 10
+    config:
+      master: false
+      data: true
+    volumeClaimTemplates:
+    - metadata:
+        name: elasticsearch-data
+      spec:
+        accessModes:
+        - ReadWriteOnce
+        resources:
+          requests:
+            storage: 1000Gi
+        storageClassName: standard
+----
+
+The Elasticsearch resource above defines two NodeSets: one for master nodes, using 10Gi volumes, and one for data nodes, using 1000Gi volumes. The Elasticsearch cluster will be composed of 13 nodes: 3 master nodes and 10 data nodes.
+
+[id="{p}-upgrading"]
+==== Upgrading the cluster
+
+ECK handles smooth upgrades from one cluster specification to another.
```

Comment:
Should we mention that we can accept new specs at any time and explain how we are always moving towards the current spec? I guess one might think we fully move to previous spec until starting moving to the new one.

Reply:
Adding `It is possible to apply a new Elasticsearch specification at any time`, which, I think, also solves https://github.com/elastic/cloud-on-k8s/pull/1859#discussion_r331408279.